frontend/rust-lib/flowy-folder/Cargo.toml

```diff
@@ -8,24 +8,29 @@ edition = "2021"
 [dependencies]
 collab = { version = "0.1.0" }
 collab-folder = { version = "0.1.0" }
+collab-document = { version = "0.1.0" }
 collab-entity = { version = "0.1.0" }
 collab-plugins = { version = "0.1.0" }
 collab-integrate = { workspace = true }
 flowy-folder-pub = { workspace = true }
 
+flowy-search.workspace = true
```

Comment:
We should introduce a new trait for dependency inversion in flowy-folder and remove the dependency `flowy-search.workspace = true` because flowy-search cannot be compiled into wasm.

You can check out the Web-CI errors

Reply:
You can try to compile the flowy-folder to wasm by adding  this in flowy-folder/cargo.toml

```rust
[lib]
crate-type = ["cdylib", "rlib"]
```

and then run
```rust
 wasm-pack build --features="web_ts"
```

Reply:
I've fixed the dependency issue (through flowy-search-pub). Do we need to implement a check in the folder managers init to _not_ index all views for wasm builds, as it's redundant (eg. `#[cfg(feature = "web_ts")]`)?